src/pages/Verify.tsx

```diff
@@ -9,16 +9,19 @@ import { LineBackground } from '../components/Backgrounds'
 
 export default function Verify() {
   const { adminId } = useParams()
-  const { put } = useFetch(`/api/admins/${adminId}/verify`)
+  const { put, response } = useFetch(`/api/admins/${adminId}/verify`)
   const [adminData, setAdminData] = useLocalStorage<AdminProps | undefined>('adminData', undefined)
   const navigate = useNavigate()
   const toast = useToast()
 
   useEffect(() => {
     put().then(
         fetchedData => {
+          if (response.ok) {
           setAdminData(fetchedData)
-          setTimeout(() => navigate('/dashboard'), 3000)
+          let redirectURL = "dashboard"
```

Comment:
You can one-line this (instead of lines 22-23):

```const redirectURL = adminData?.fullyRegistered ? '/dashboard' : `/verify/${adminData?.id}/verifyCollab` ```